spec/unit/knife/bootstrap_windows_winrm_spec.rb

```diff
@@ -41,6 +42,109 @@
 
   let(:initial_fail_count) { 4 }
 
+  context "knife secret-file && knife secret options are passed" do
```

Comment:
It might be better to move this test under functional rather than unit since it makes changes to the file system. If everything can be fully mocked out, then keeping it a unit test is fine.